CryptoPkg/Library/BaseCryptLibMbedTls/Pk/CryptPkcs7VerifyCommon.c

```diff
@@ -414,14 +479,45 @@ MbedtlsPkcs7SignedDataVerifySigners (
   )
 {
   INT32 Ret;
-  UINT8 Hash[32];
+  UINT8 Hash[MBEDTLS_MD_MAX_SIZE];
   mbedtls_pk_context Pk;
   CONST mbedtls_md_info_t *MdInfo;
+  INTN HashLen;
 
   Pk = Cert->pk;
+
+  //all the hash algo
+  MdInfo = mbedtls_md_info_from_type (MBEDTLS_MD_SHA1);

```

Comment:
why you still have SHA1? Is this absolutely needed?

Reply:
Thanks. I have deleted the SHA1 in Pkcs7.